Assets/Scripts/CandyAnimationsController.cs

```diff
@@ -0,0 +1,91 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CandyAnimationsController : MonoBehaviour
+{
+    private CandyPool pool;
+    private static CandyAnimationsController instance;
+    public static CandyAnimationsController Instance
+    {
+        get
+        {
+            if (instance == null)
+            {
+                GameObject go = new GameObject("AnimationsController");
+                instance = go.AddComponent<CandyAnimationsController>();
+                instance.pool = FindAnyObjectByType<CandyPool>();
+            }
+            return instance;
+        }
+    }
+
+    
+    public IEnumerator MoveCandy(GameObject candy, Vector3 startPosition, Vector3 targetPosition, float duration)
+    {
+        //Vector3 startPosition = candy.transform.position;
+        float elapsedTime = 0;
+
+        while (elapsedTime < duration)
+        {
+            candy.transform.position = Vector3.Lerp(startPosition, targetPosition, (elapsedTime/duration));
+            elapsedTime += Time.deltaTime;
+            yield return null;
+        }
+        candy.transform.position = targetPosition;
+    }
+    public IEnumerator RotateMatchingCandies(GameObject candy, float duration, int rotations)
+    {
+        float timePerRotation = duration / rotations;
+
+            for (int i = 0; i < rotations; i++)
+            {
+                float elapsedTime = 0f;
+                float startRotation = candy.transform.rotation.eulerAngles.z;
+                float endRotation = startRotation + 360;
+
+
+                while (elapsedTime < timePerRotation)
+                {
+                    float zRotation = Mathf.Lerp(startRotation, endRotation, (elapsedTime / duration));
+                    candy.transform.rotation = Quaternion.Euler(0, 0, zRotation);
+                    elapsedTime += Time.deltaTime;
+                    //If the method were void and you didn't yield, all the calculations would be executed in a single frame.
+                    
+
+                    yield return null;
+                }
+                candy.transform.rotation = Quaternion.Euler(0, 0, endRotation);
+
+            }
+        pool.ReturnCandy(candy);
+    }
+    public List<GameObject> CreateRotationList( List<GameObject> matchesHor, List<GameObject> matchesVer, GameSettings gameSettings, CandyPool candyPool)
+    {
+        List<GameObject> combinedLists = new List<GameObject>();
```

Comment:
One optimization tip:

Whenever you are using a "new" keyword, or in other words you initialize a class (in this case initializing the list) you are allocating new memory and it forces the previous allocation to be removed by the garbage collector which is CPU intensive. What you can do here instead, is to use a global array that is defined in your class, initialize it in "Awake" method and here in this line, you can just call `combinedLists.Clear()` and `clonedLists.Clear()`